### PR TITLE
update MFEM, superlu-dist, and Tribol

### PR DESCRIPTION
**Pull request by ebchin**
Contact support needs this recent fix: https://github.com/mfem/mfem/commit/71fab657ec8c4a25c761338d9df3e0cdf5faf24e and will benefit from this recently merged PR: https://github.com/mfem/mfem/pull/3557 in MFEM.

This more recent version of MFEM does not need an old superlu-dist, so the version specified in the serac `package.py` has been updated.

Also, Tribol has been updated to the most recent develop.

### CODE REVIEW

**Comment by ebchin** [comment]
I think this one is ready to go now

**Review by chapman39 — APPROVED** [review]

File: scripts/spack/packages/serac/package.py [hunk 1]
```diff
@@ -137,8 +137,6 @@ class Serac(CachedCMakePackage, CudaPackage):
     depends_on("adiak+mpi", when="+profiling")
     depends_on("caliper+mpi+adiak~papi", when="+profiling")
 
-    depends_on("superlu-dist@6.1.1")
```
Comment (jamiebramwell) [inline scripts/spack/packages/serac/package.py:140]: @white238 @chapman39 , do you remember why this was here? I can't figure out why we need it, but I'm sure there was a reason.

**Review by jamiebramwell — APPROVED** [review]

File: scripts/spack/packages/serac/package.py (hunk 1 above)
Comment (chapman39) [reply scripts/spack/packages/serac/package.py:140]: looking at the blame, it was added in this pr https://github.com/LLNL/serac/pull/764, its not clear why though

edit: this was added years ago actually

Comment (ebchin) [reply scripts/spack/packages/serac/package.py:140]: Looks like the version number was added in [this commit](https://github.com/LLNL/serac/commit/14de70691423bd6d5c2fbbe7617f1e591bd3f24a) to lock down package version numbers.  With that in mind, I'll add it back and update the version to the one spack decided to build (8.1.2).

File: host-configs/docker/clang@10.0.0.cmake [hunk 2]
```diff
@@ -4,9 +4,9 @@
...
 
-set(CMAKE_BUILD_TYPE "Debug" CACHE STRING "")
+set(CMAKE_BUILD_TYPE "Release" CACHE STRING "")
```
Comment (ebchin) [inline host-configs/docker/clang@10.0.0.cmake:9]: @white238 @chapman39 Should the docker builds be changed back to debug?

Comment (chapman39) [reply host-configs/docker/clang@10.0.0.cmake:9]: i think they should be release but lets see what chris says

**Pull request merged**